### PR TITLE
Add SignInConfirmTotpCode screen

### PR DESCRIPTION
**Pull request by mattcreaser**
- [x] PR conforms to [Pull Request](https://github.com/aws-amplify/amplify-ui-android/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guidelines.

*Issue #, if available:*

*Description of changes:*
- Adds the SignInConfirmTotpCode screen to the totp feature branch.

*How did you test these changes?*
- Unit test
- Screenshot test
- Code can be used to sign in when there is an authenticator registered and MFA is enabled

*Documentation update required?*
- [x] No
- [ ] Yes (Please include a PR link for the documentation update)

*General Checklist*
- [x] Added Unit Tests
- [ ] Added Integration Tests
- [ ] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: authenticator/src/main/java/com/amplifyframework/ui/authenticator/states/SignInConfirmTotpCodeStateImpl.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    init {
+        form.addFields {
+            confirmationCode()
+        }
+    }
+
+    override fun moveTo(step: AuthenticatorInitialStep) = onMoveTo(step)
+
+    override suspend fun confirmSignIn() = doSubmit {
+        val confirmationCode = form.getTrimmed(FieldKey.ConfirmationCode)!!
```
Comment (tylerjroach) [inline authenticator/src/main/java/com/amplifyframework/ui/authenticator/states/SignInConfirmTotpCodeStateImpl.kt:21]: Just double checking !! is ok in this case.

File: authenticator/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -17,4 +17,6 @@
     <!-- Confirmation Code Screen -->
     <string name="amplify_ui_authenticator_confirmation_code_sent">A confirmation code has been sent to <b>%s</b>.</string>
     <string name="amplify_ui_authenticator_confirmation_code_sent_unknown">A confirmation code has been sent.</string>
+    <!-- TOTP Code Screen -->
+    <string name="amplify_ui_authenticator_enter_totp_code">Please enter the code from your registered Authenticator app</string>
```
Comment (tylerjroach) [inline authenticator/src/main/res/values/strings.xml:21]: FYI, just wanted to call this out. We had changes that needed to be made on the liveness side and we determined that string translation additions are not considered breaking changes. With each version bump, the customer should ensure that if translations are required, they evaluate any string changes. We do need to make sure to add this (and the other string change, to the changelog).

File: authenticator/src/main/res/values/titles.xml [hunk 3]
```diff
@@ -22,4 +22,5 @@
     <string name="amplify_ui_authenticator_title_password_reset">Reset Password</string>
     <string name="amplify_ui_authenticator_title_signup_confirm">Enter your confirmation code</string>
     <string name="amplify_ui_authenticator_title_verify_user">Account recovery requires verified contact information</string>
+    <string name="amplify_ui_authenticator_title_signin_confirm_totp">Enter your one-time passcode</string>
```
Comment (tylerjroach) [inline authenticator/src/main/res/values/titles.xml:25]: Making note to ensure we call this out in changelog.

**Review by tylerjroach — APPROVED** [review]
Great work.

File: authenticator/src/main/java/com/amplifyframework/ui/authenticator/states/SignInConfirmTotpCodeStateImpl.kt (hunk 1 above)
Comment (mattcreaser) [reply authenticator/src/main/java/com/amplifyframework/ui/authenticator/states/SignInConfirmTotpCodeStateImpl.kt:21]: Yes this is how all the `*StateImpl` classes are implemented. The field is added in the init function on L14 so we know it's present.

**Review by ankpshah — COMMENTED** [review]
PR looks good to me. Clean and Efficient !

**Pull request merged**